### PR TITLE
Introduce new cli to deploy the KubeVirt cluster and HCO

### PR DESCRIPTION
**Pull request by nunnatsa**
Add new `hcocli` executable tool to replace the deploy/deploy.sh script

In the current version, the tool deploys all the dependencies and the HyperConverged custom resource.

hcocli help:
```
Usage:
  hcocli [flags]
  hcocli [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  deploy      Deploy the HyperConverged custom resource
  help        Help about any command
  install     Install all the dependencies, the required resources and the kubevirt hyperconverged cluster operator

Flags:
  -h, --help                help for hcocli
      --kubeconfig string   path of the kubeconfig file
      --timeout duration    timeout (default 20m0s)

Use "hcocli [command] --help" for more information about a command.
```

**What this PR does / why we need it**:

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
None
```

**Release note**:

```release-note
Introduce a new cli to deploy the KubeVirt cluster and HCO
```

### CODE REVIEW

File: cmd/hcocli/deploycr/deploy.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(cmd.Context(), timeout)
+	defer cancel()
+	cmd.SetContext(ctx)
+
+	hc := &v1beta1.HyperConverged{}
+
+	bytes, err := dep.deploy.ReadFile("deploy/hco.cr.yaml")
```
Comment (tiraboschi) [inline cmd/hcocli/deploycr/deploy.go:62]: why should we read it from the file?
if it's just to have consistent defaults we can use something like:
```
	defaultScheme := runtime.NewScheme()
	_ = hcov1beta1.AddToScheme(defaultScheme)
	_ = hcov1beta1.RegisterDefaults(defaultScheme)
	defaultHco := &hcov1beta1.HyperConverged{
		TypeMeta: metav1.TypeMeta{
			APIVersion: util.APIVersion,
			Kind:       util.HyperConvergedKind,
		},
		ObjectMeta: metav1.ObjectMeta{
			Name: crName,
		}}
	defaultScheme.Default(defaultHco)
```

File: cli.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+		Run: func(cmd *cobra.Command, args []string) {
+			_ = cmd.Help()
+			os.Exit(1)
+		},
+	}
+
+	timeout := time.Minute * 20
+	cmd.PersistentFlags().DurationVar(&timeout, "timeout", timeout, "timeout")
+	cmd.PersistentFlags().String("kubeconfig", "", "path of the kubeconfig file")
```
[the remark this answers is not in the document]
Comment (nunnatsa) [reply cli.go:35]: KUBECONFIG is used here internally in `client.GetClient`. this flag allow the user to override the kubeconfig, as in kubectl etc.

**Comment by nunnatsa** [comment]
/hold

rethinking about the whole thing

**Comment by nunnatsa** [comment]
Probably not the best idea. Closing

**Pull request closed without merging**